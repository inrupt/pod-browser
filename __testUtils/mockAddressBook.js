/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import {
  addStringNoLocale,
  addUrl,
  createSolidDataset,
  mockSolidDatasetFrom,
  mockThingFrom,
  setThing,
} from "@inrupt/solid-client";
import { acl, dc, rdf } from "rdf-namespaces";
import { chain } from "../src/solidClientHelpers/utils";
import { aliceWebIdUrl } from "./mockPersonResource";
import { vcardExtras } from "../src/addressBook";
import { getAddressBookIndexDefaultUrl } from "../src/models/addressBook";
import { joinPath } from "../src/stringHelpers";

export const defaultContainerUrl = "https://user.example.com/contacts";
export const defaultAddressBookDatasetUrl = joinPath(
  defaultContainerUrl,
  "index.ttl"
);
export const defaultAddressBookThingUrl = `${defaultAddressBookDatasetUrl}#this`;

export function mockAddressBookThing(options = {}) {
  const containerUrl = options.containerUrl || defaultContainerUrl;
  const owner = options.owner || aliceWebIdUrl;
  const indexUrl =
    options.indexUrl || getAddressBookIndexDefaultUrl(containerUrl);
  const mainIndexUrl = `${indexUrl}#this`;
  return chain(
    mockThingFrom(mainIndexUrl),
    (t) => addUrl(t, rdf.type, vcardExtras("AddressBook")),
    (t) => addUrl(t, acl.owner, owner),
    (t) => addStringNoLocale(t, dc.title, "Contacts")
  );
}

export function mockAddressBookDataset(
  options = {},
  thing = mockAddressBookThing(options)
) {
  const containerUrl = options.containerUrl || defaultContainerUrl;
  const indexUrl =
    options.indexUrl || getAddressBookIndexDefaultUrl(containerUrl);
  return chain(mockSolidDatasetFrom(indexUrl), (d) => setThing(d, thing));
}

export default function mockAddressBook(options = {}) {
  const containerUrl = options.containerUrl || defaultContainerUrl;
  const thing = mockAddressBookThing(options);
  return {
    containerUrl,
    thing,
    dataset: mockAddressBookDataset(options, thing),
  };
}

export function mockEmptyAddressBook(containerUrl, owner, title = "Contacts") {
  const thing = chain(
    mockAddressBookThing({ containerUrl, owner }),
    (t) => addUrl(t, rdf.type, vcardExtras("AddressBook")),
    (t) => addUrl(t, acl.owner, owner),
    (t) => addStringNoLocale(t, dc.title, title)
  );
  return {
    containerUrl,
    dataset: setThing(createSolidDataset(), thing),
    thing,
  };
}
