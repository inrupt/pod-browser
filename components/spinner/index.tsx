import React, { ReactElement } from "react";
import { CircularProgress, Container } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(() => ({
  "MuiCicularProgress-root": {
    display: "block",
    position: "fixed",
    top: "50%",
    right: "50%",
  },
}));

export default function Spinner(): ReactElement {
  const classes = useStyles();
  return (
    <Container>
      <CircularProgress
        className={classes["MuiCicularProgress-root"]}
        color="primary"
      />
    </Container>
  );
}
