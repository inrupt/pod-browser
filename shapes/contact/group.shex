PREFIX vcard: <http://www.w3.org/2006/vcard/ns#>
PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>

# This shape describes some of the information we store when adding a group to
# an address book. The resource it is used for is to describe the group itself
# (meaning the resource should only contain one group). We use the id "#this"
# to refer to the subject of the group.

# This model was originally developed as part of the
# SolidOS/Data Browser project, and we strive to keep it interoperable with
# their work. Some custom terms have been introduced to vocabularies we think
# they could belong to (e.g. vcard:includesGroup). We'll make a note of the
# custom term when used.

vcard:Group {
  vcard:fn xsd:string {1} ;
  vcard:note xsd:string ? ;
  vcard:hasMember vcard:Individual * OR vcard:Organization * ;
  # vcard:includesGroup is a custom term
  # vcard:AddressBook is a custom class
  ^vcard:includesGroup vcard:AddressBook
}

# Note that we also cache the name of each member, so that we don't have to
# fetch the profile of each member to get their name. An example of this can
# be seen in example/Group/Group_1.ttl.
