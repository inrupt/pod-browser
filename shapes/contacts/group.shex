PREFIX vcard: <http://www.w3.org/2006/vcard/ns#>
PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>

vcard:Group {
  vcard:fn xsd:string {1} ;
  vcard:note xsd:string ? ;
  vcard:hasMember vcard:Individual * OR vcard:Organization * ;
  ^vcard:includesGroup vcard:AddressBook
}
