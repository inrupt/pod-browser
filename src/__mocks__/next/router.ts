export default {
  push: jest.fn().mockResolvedValue(null),
};
