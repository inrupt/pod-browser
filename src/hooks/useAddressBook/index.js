/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { useContext, useEffect, useState } from "react";
import { fetchLitDataset } from "@inrupt/solid-client";
import SessionContext from "../../contexts/sessionContext";
import useAuthenticatedProfile from "../useAuthenticatedProfile";
import { contactsContainerIri, saveNewAddressBook } from "../../addressBook";
import { getResource } from "../../solidClientHelpers/resource";
import { ERROR_CODES, isHTTPError } from "../../solidClientHelpers/utils";

export default function useAddressBook() {
  const [addressBook, setAddressBook] = useState(null);
  const [error, setError] = useState(null);
  const { session } = useContext(SessionContext);
  const profile = useAuthenticatedProfile();

  useEffect(() => {
    if (!session.info.isLoggedIn || !profile) return;
    const { pods, webId } = profile;
    const { fetch } = session;
    const contactsIri = contactsContainerIri(pods[0]);

    (async () => {
      let addressBookUri;

      const {
        response: existingAddressBook,
        error: existingError,
      } = await getResource(contactsIri, fetch);

      if (existingAddressBook) {
        setAddressBook(existingAddressBook.dataset);
        return;
      }

      if (existingError && !isHTTPError(existingError, ERROR_CODES.NOT_FOUND)) {
        const {
          response: newAddressBook,
          error: newError,
        } = await saveNewAddressBook(
          {
            iri: contactsIri,
            owner: webId,
          },
          fetch
        );
        if (newError) {
          setError(newError);
          return;
        }
        addressBookUri = newAddressBook.iri;
      } else if (existingError) {
        setError(existingError);
        return;
      }

      setAddressBook(await fetchLitDataset(addressBookUri, { fetch }));
    })();
  }, [session, profile]);

  return [addressBook, error];
}
