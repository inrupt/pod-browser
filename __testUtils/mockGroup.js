/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import {
  addUrl,
  getStringNoLocale,
  getThing,
  mockSolidDatasetFrom,
  mockThingFrom,
  setStringNoLocale,
  setThing,
  setUrl,
} from "@inrupt/solid-client";
import { rdf, vcard } from "rdf-namespaces";
import { chain } from "../src/solidClientHelpers/utils";
import { createGroupDatasetUrl } from "../src/models/group";
import { vcardExtras } from "../src/models/addressBook";

function mockGroupThing(name, groupThingUrl) {
  return chain(
    mockThingFrom(groupThingUrl),
    (t) => setUrl(t, rdf.type, vcard.Group),
    (t) => setStringNoLocale(t, vcard.fn, name)
  );
}

function mockIndexThing(addressBook, groupThingUrl) {
  return setUrl(addressBook.thing, vcardExtras("includesGroup"), groupThingUrl);
}

export function addGroupToIndexDataset(
  dataset,
  addressBook,
  name,
  groupThingUrl
) {
  return chain(
    dataset,
    (d) => setThing(d, mockGroupThing(name, groupThingUrl)),
    (d) => setThing(d, mockIndexThing(addressBook, groupThingUrl))
  );
}

// export function addGroupsToAddressBook(addressBook, groupsToAdd) {
//   const { groups, ...rest } = addressBook;
//   return {
//     ...rest,
//     groups: {
//       iri: addressBook.groups.iri,
//       dataset: chain(
//         addressBook.groups.dataset,
//         ...groupsToAdd.map(({ dataset, iri }) => (d) => {
//           const group = getThing(dataset, iri);
//           const name = getStringNoLocale(group, vcard.fn);
//           return setThing(d, mockGroupThing(name, iri));
//         }),
//         ...groupsToAdd.map((group) => (d) =>
//           setThing(d, mockIndexThing(addressBook, group.iri))
//         )
//       ),
//     },
//   };
// }

export default function mockGroup(
  addressBook,
  name,
  { url, id, members } = {}
) {
  const groupDatasetUrl = url || createGroupDatasetUrl(addressBook, id);
  const groupThingUrl = `${groupDatasetUrl}#this`;
  const groupThing = chain(
    mockGroupThing(name, groupThingUrl),
    ...(members || []).map((agentUrl) => (t) =>
      addUrl(t, vcard.hasMember, agentUrl)
    )
  );
  return {
    dataset: chain(
      mockSolidDatasetFrom(groupDatasetUrl),
      (d) => setThing(d, groupThing),
      (d) => setThing(d, mockIndexThing(addressBook, groupThingUrl))
    ),
    thing: groupThing,
  };
}
