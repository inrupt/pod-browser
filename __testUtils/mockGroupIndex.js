/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { mockSolidDatasetFrom } from "@inrupt/solid-client";
import { getContactIndexDefaultUrl } from "../src/models/contact/collection";
import { addGroupToMockedIndexDataset } from "./mockGroupContact";
import { getGroupName, getGroupUrl } from "../src/models/group";
import { chain } from "../src/solidClientHelpers/utils";
import { GROUP_CONTACT } from "../src/models/contact/group";

export default function mockGroupIndex(addressBook, groups) {
  const contactIndexUrl = getContactIndexDefaultUrl(
    addressBook.containerUrl,
    GROUP_CONTACT
  );
  return {
    dataset: chain(
      mockSolidDatasetFrom(contactIndexUrl),
      ...groups.map((g) => (d) =>
        addGroupToMockedIndexDataset(
          d,
          addressBook,
          getGroupName(g),
          getGroupUrl(g)
        )
      )
    ),
    type: GROUP_CONTACT,
  };
}
