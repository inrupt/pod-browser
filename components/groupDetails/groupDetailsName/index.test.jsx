/**
 * Copyright 2020 Inrupt Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal in
 * the Software without restriction, including without limitation the rights to use,
 * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
 * Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import React from "react";
import { useRouter } from "next/router";
import { useSession } from "@inrupt/solid-ui-react";
import userEvent from "@testing-library/user-event";
import { waitFor } from "@testing-library/react";
import useAddressBook from "../../../src/hooks/useAddressBook";
import useContacts from "../../../src/hooks/useContacts";
import useGroup from "../../../src/hooks/useGroup";
import mockGroup from "../../../__testUtils/mockGroup";
import mockAddressBook from "../../../__testUtils/mockAddressBook";
import renderGroupsPage from "../../../__testUtils/renderGroupsPage";
import GroupDetailsName, {
  MESSAGE_GROUP_DETAILS_NAME_REQUIRED,
  TESTCAFE_ID_GROUP_DETAILS_NAME,
  TESTCAFE_ID_GROUP_DETAILS_NAME_BUTTON,
  TESTCAFE_ID_GROUP_DETAILS_NAME_FIELD,
} from "./index";
import { TESTCAFE_ID_SPINNER } from "../../spinner";
import { TESTCAFE_ID_ERROR_MESSAGE } from "../../errorMessage";
import * as contactGroupFns from "../../../src/models/contact/group";
import mockGroupIndex from "../../../__testUtils/mockGroupIndex";
import { getContactAllFromContactsIndex } from "../../../src/models/contact";

jest.mock("../../../src/hooks/useAddressBook");
const mockedAddressBookHook = useAddressBook;

jest.mock("../../../src/hooks/useContacts");
const mockedContactsHook = useContacts;

jest.mock("../../../src/hooks/useGroup");
const mockedGroupHook = useGroup;

jest.mock("@inrupt/solid-ui-react");
const mockedSessionHook = useSession;

jest.mock("next/router");
const mockedRouterHook = useRouter;

const addressBook = mockAddressBook();
const fetch = jest.fn();

const group1Name = "Group 1";
const group1Url = "http://example.com/group1.ttl#this";
const group1 = mockGroup(group1Name, group1Url);
const updatedGroup = mockGroup("New name", group1Url);
const updatedGroupIndex = mockGroupIndex(addressBook, [updatedGroup]);

describe("GroupDetailsName", () => {
  let mutateAddressBook;
  let mutateGroups;
  let mutateGroup;

  beforeEach(() => {
    mutateAddressBook = jest.fn();
    mutateGroups = jest.fn();
    mutateGroup = jest.fn();

    mockedAddressBookHook.mockReturnValue({
      data: addressBook,
      error: null,
      mutate: mutateAddressBook,
    });
    mockedContactsHook.mockReturnValue({
      data: [group1],
      error: null,
      mutate: mutateGroups,
    });
    mockedGroupHook.mockReturnValue({
      data: group1,
      error: null,
      mutate: mutateGroup,
    });
    mockedSessionHook.mockReturnValue({ fetch });
    mockedRouterHook.mockReturnValue({ query: { iri: group1Url } });
  });

  it("renders a form", () => {
    const { asFragment, getByTestId } = renderGroupsPage(<GroupDetailsName />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId(TESTCAFE_ID_GROUP_DETAILS_NAME)).toBeDefined();
    expect(getByTestId(TESTCAFE_ID_GROUP_DETAILS_NAME_FIELD)).toBeDefined();
    expect(getByTestId(TESTCAFE_ID_GROUP_DETAILS_NAME_FIELD).value).toEqual(
      group1Name
    );
    expect(getByTestId(TESTCAFE_ID_GROUP_DETAILS_NAME_FIELD)).not.toBe(
      document.activeElement
    );
  });

  it("sets focus on input field when newly created", () => {
    mockedRouterHook.mockReturnValue({
      query: { iri: group1Url, created: "" },
    });
    const { getByTestId } = renderGroupsPage(<GroupDetailsName />);
    expect(getByTestId(TESTCAFE_ID_GROUP_DETAILS_NAME_FIELD)).toBe(
      document.activeElement
    );
  });

  it("renders a spinner while loading group", () => {
    mockedGroupHook.mockReturnValue({});
    const { getByTestId } = renderGroupsPage(<GroupDetailsName />);
    expect(getByTestId(TESTCAFE_ID_SPINNER)).toBeDefined();
  });

  it("renders an error if something goes wrong", () => {
    const errorMessage = "error";
    mockedContactsHook.mockReturnValue({ error: new Error(errorMessage) });
    const { getByTestId } = renderGroupsPage(<GroupDetailsName />);
    expect(getByTestId(TESTCAFE_ID_ERROR_MESSAGE)).toBeDefined();
  });

  describe("submitting the form", () => {
    let mockedRenameGroup;

    beforeEach(() => {
      mockedRenameGroup = jest
        .spyOn(contactGroupFns, "renameGroup")
        .mockResolvedValue({
          group: updatedGroup,
          groupIndex: mockGroupIndex(addressBook, [updatedGroup]),
        });
    });

    it("updates the name of the group", async () => {
      const { getByTestId } = renderGroupsPage(<GroupDetailsName />);
      userEvent.type(
        getByTestId(TESTCAFE_ID_GROUP_DETAILS_NAME_FIELD),
        "{selectall}{del}New name{enter}"
      );

      await waitFor(() => {
        expect(
          getByTestId(TESTCAFE_ID_GROUP_DETAILS_NAME_BUTTON)
        ).toBeDefined();
      });
      expect(mockedRenameGroup).toHaveBeenCalledWith(
        addressBook,
        group1,
        "New name",
        fetch
      );
      expect(mutateGroup).toHaveBeenCalledWith(updatedGroup);
      expect(mutateGroups).toHaveBeenCalledWith(
        getContactAllFromContactsIndex(updatedGroupIndex)
      );
    });

    it("renders a spinner while processing", async () => {
      const { getByTestId } = renderGroupsPage(<GroupDetailsName />);
      userEvent.type(
        getByTestId(TESTCAFE_ID_GROUP_DETAILS_NAME_FIELD),
        "{selectall}{del}New name{enter}"
      );

      expect(getByTestId(TESTCAFE_ID_SPINNER)).toBeDefined();

      await waitFor(() => {
        expect(
          getByTestId(TESTCAFE_ID_GROUP_DETAILS_NAME_BUTTON)
        ).toBeDefined();
      });
    });

    it("won't allow empty names", () => {
      const { getByTestId, getByText } = renderGroupsPage(<GroupDetailsName />);
      userEvent.type(
        getByTestId(TESTCAFE_ID_GROUP_DETAILS_NAME_FIELD),
        "{selectall}{del}{enter}"
      );
      expect(getByText(MESSAGE_GROUP_DETAILS_NAME_REQUIRED)).toBeDefined();
      expect(mockedRenameGroup).not.toHaveBeenCalled();
    });
  });
});
