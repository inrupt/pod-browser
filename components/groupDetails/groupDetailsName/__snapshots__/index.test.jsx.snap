// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupDetailsName renders a form 1`] = `
<DocumentFragment>
  <form
    class="PodBrowser-form PodBrowser-group-details-name"
    data-testid="group-details-name"
  >
    <div
      class="PodBrowser-input-group"
    >
      <label
        class="PodBrowser-label"
        for="GroupName"
      >
        Group Name
      </label>
      <div
        class="PodBrowser-input-group__inner PodBrowser-input-group__inner--with-button"
      >
        <input
          class="PodBrowser-input"
          data-testid="group-details-name-field"
          id="GroupName"
          required=""
          type="text"
          value="Group 1"
        />
        <button
          class="PodBrowser-button PodBrowser-button--with-input"
          data-testid="group-details-name-button"
          type="submit"
        >
          <span
            class="PodBrowser-button__text"
          >
            Save
          </span>
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
